inference-engine/src/low_precision_transformations/src/assign_and_read_value.cpp

```diff
@@ -0,0 +1,142 @@
+#include <ngraph/ngraph.hpp>
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "low_precision/network_helper.hpp"
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <openvino/op/util/assign_base.hpp>
+#include <queue>
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::low_precision::AssignAndReadValueTransformation, "AssignAndReadValueTransformation", 0);
+
+AssignAndReadValueTransformation::AssignAndReadValueTransformation(const std::shared_ptr<ngraph::Function> function, const Params& params) :
+    LayerTransformation(params), function(function) {
+    auto assign3 = pattern::wrap_type<opset3::Assign>({ pattern::wrap_type<opset1::Multiply>() });
+    auto assign6 = pattern::wrap_type<opset6::Assign>({ pattern::wrap_type<opset1::Multiply>() });
+
+    ngraph::graph_rewrite_callback callback = [=](pattern::Matcher& m) {
+        const auto& opsMap = m.get_pattern_value_map();
+        auto op = m.get_match_root();
+        auto assignIt = opsMap.find(assign3);
+        if (assignIt == opsMap.end()) {
+            assignIt = opsMap.find(assign6);
+        }
+        const auto assign = assignIt->second.get_node()->shared_from_this();
+        // check that we have ReadValue as the first dependency
+        if (assign->get_control_dependencies().empty()) {
+            return false;
+        }
+        const auto readValue = std::dynamic_pointer_cast<op::ReadValueBase>(assign->get_control_dependencies()[0]);
+        if (!readValue) {
+            return false;
+        }
+
+        // TODO: remove this limitation and change the transformation when this constant will be accepted to be non-zero
+        if (!NetworkHelper::isZeroConst(readValue->get_input_node_shared_ptr(0))) {
+            return false;
+        }
+
+        if (transformation_callback(op)) {
+            return false;
+        }
+        return transform(*context, m);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(
+        std::make_shared<pattern::op::Or>(OutputVector{ assign3, assign6 }),
+        "AssignAndReadValueTransformation");
+    this->register_matcher(m, callback);
+}
+
+bool AssignAndReadValueTransformation::transform(TransformationContext& context, ngraph::pattern::Matcher& m) {
+    if (!canBeTransformed(context, m.get_match_root())) {
+        return false;
+    }
+
+    const auto oldAssign = m.get_match_root();
+    const auto readValue = oldAssign->get_control_dependencies()[0];
+    oldAssign->remove_control_dependency(readValue);
+
+    const auto assign = NetworkHelper::separateInStandaloneBranch(oldAssign);
+    const auto dequantization = NetworkHelper::getDequantization(assign);
+
+    auto oldVar = ov::as_type_ptr<op::ReadValueBase>(readValue)->get_variable();
+    auto variable_info = oldVar->get_info();
+    oldVar->update({variable_info.data_shape, dequantization.data.get_element_type(), variable_info.variable_id});
```

Comment:
It's not so obvious why we are changing varInfo here, but not using oldVar or varInfo after
please left a comment here

Reply:
@itikhono This is shared_ptr, so it's changed as dependency for ReadValue to set new precision for input default constant.

Reply:
Yes, I got it, but I guess we need a comment in the code :)
Something like that:
 We set a new precision for input default constant here because **old**Var it's copied in copy_with_new_inputs method and actually affects **new**ReadValue.